### PR TITLE
Determine workflow run id more robustly to avoid race conditions

### PR DESCRIPTION
**Pull request by neilmayhew**
This builds on #34

### CODE REVIEW

**Comment by neilmayhew** [comment]
If two runs are started very close together, this logic will see two new runs. It's impossible to tell which is the one we triggered, so we wait for both of them. If either of them fails (and we're propagating failures) we fail this one.

File: entrypoint.sh [hunk 1]
```diff
@@ -82,64 +88,96 @@ validate_args() {
...
+  SINCE=$(date -u -Iseconds -d "@$((START_TIME - 120))") # Two minutes ago, to overcome clock skew
+
+  OLD_RUNS=$(get_workflow_runs "$SINCE")
+
+  echo >&2 "Triggering workflow:"
+  echo >&2 "  workflows/${INPUT_WORKFLOW_FILE_NAME}/dispatches"
+  echo >&2 "  {\"ref\":\"${ref}\",\"client_payload\":${client_payload}}"
+
+  api "workflows/${INPUT_WORKFLOW_FILE_NAME}/dispatches" \
+    --data "{\"ref\":\"${ref}\",\"inputs\":${client_payload}}"
```
Comment (NotNils) [inline entrypoint.sh:133]: Should the API call here match the string that is echoed out above it?

Comment (neilmayhew) [reply entrypoint.sh:133]: Good catch! The commits this is based on (from #34) changed the action input name from `inputs` to `client_payload` but the JSON key here is defined by the `dispatches` API and needs to be `inputs`. I'll change the `echo` call.

**Review by keithconvictional — APPROVED** [review]
LGTM

**Comment by duhow** [comment]
Changes merged here reverted #41 😭

**Comment by neilmayhew** [comment]
@duhow That's unfortunate. Looks like some of those changes got missed while resolving merge conflicts. The most important one is the `-c` option to `jq`, I think.

Do you plan to submit a new PR?

**Comment by keithconvictional** [comment]
@duhow @neilmayhew - Sorry about that. If you put a new PR, I'll review by end of week. The slow down was testing, and I have a better way at testing these changes now.

**Comment by neilmayhew** [comment]
@keithconvictional I've pushed a new PR (#47)

**Pull request merged**